Library/TeamTalkLib/avstream/WebRTCPreprocess.cpp

```diff
@@ -97,6 +93,17 @@ int WebRTCPreprocess(webrtc::AudioProcessing& apm, const media::AudioFrame& infr
                                            const_cast<int16_t*>(&infrm.output_buffer[in_index * echo_cfg.num_channels()]));
             MYTRACE_COND(ret != webrtc::AudioProcessing::kNoError,
                          ACE_TEXT("WebRTC failed to process reverse audio frame. Result: %d\n"), ret);
+
+            // As per audio_processing.h, set_stream_delay_ms should be called after ProcessReverseStream
+            int ret = apm.set_stream_delay_ms(delayms);
+
+            MYTRACE_COND(ret != webrtc::AudioProcessing::kNoError,
+                ACE_TEXT("WebRTC failed to set stream delay for echo cancellation. Result: %d, delay: %d\n"), ret, infrm.duplex_callback_delay);
+            if (ret != webrtc::AudioProcessing::kNoError)
+                return -1;
+
+            //WebRTC is fed in 10ms frames, so increment the delay 10ms for every WebRTC frame
+            delayms += 10;
```

Comment:
I don't quite understand this increment of 10. Isn't the set_stream_delay_ms() function an offset? Shouldn't all 10msec frames have the same offset?

Reply:
Yes, you're right. This is simply a mistake on my part. I'll remove the +10.
I'm chasing a gradually failing AEC and tried this.